pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Topic.java

```diff
@@ -302,6 +302,11 @@ CompletableFuture<? extends TopicStatsImpl> asyncGetStats(boolean getPreciseBack
      */
     CompletableFuture<SchemaVersion> deleteSchema();
 
+    /**
+     * Delete the schema if this topic has a schema defined for it.
+     */
+    CompletableFuture<SchemaVersion> deleteSchemaIfNonPartitioned();
```

Comment:
We can override `deleteSchema` in PersistentTopic

Reply:
Changed